mempool/src/shared_mempool/peer_manager.rs

```diff
@@ -185,11 +178,13 @@ impl PeerManager {
             return;
         };
 
-        // Only broadcast to peer that is both alive and picked.
-        if !state.is_alive || !self.is_picked_peer(&peer) {
+        // Only broadcast to peers that are alive.
+        if !state.is_alive {
             return;
         }
 
+        // TODO: Use `prioritized_peers` to determine which to send to.  Currently, always sends to all
```

Comment:
I'm guessing this is where you'll be going to use this list of prioritized peers? Other than here, I can't see it being used yet 😄

Reply:
It's a future optimization, don't send to all the peers but only the first X.

The important thing to remember here is the point of the `prioritized peers` is to get the `primary upstream` that was previously used in logic here.  Additionally, this logic right now makes it **much** easier as there is no real differentiation between the Validator and non-Validator use case.  It's handled by who you're already connected to, and **then** you can choose who you want to send to (primary for a VFN always being the validator)